main.cpp

```diff
@@ -602,8 +602,9 @@ static bool add_endpoints(Mainloop &mainloop)
             return false;
         }
 
-        g_endpoints[i++] = udp.release();
-        mainloop.add_fd(udp->fd, g_endpoints[i], EPOLLIN);
+        g_endpoints[i] = udp.release();
```

Comment:
ugh. thanks. I fixed this locally and forgot to amend before pushing.
Could you send this fix independently?

Also,  `- Using unique_ptr after release()` is not true... the problem is that we would pass g_endpoints[i+1] to add_fd due to the second issue you pointed out.